STYLE-GUIDE.md

```diff
@@ -0,0 +1,134 @@
+the first is a set of rules that every contribution may be tested against and mechanisms for tracking the 
+project's compliance automatically. The second expands upon some of the intentions behind the OpenLMIS principles 
+to influence better design and implementation of OpenLMIS features.
+
+Often the state of the code-base is far from the outlined ideals.  This presents us with the opportunity to improve 
+upon the current state of the project while recording lessons learned here and through the developer forums.  In 
+that vein, this guide is in no way meant to be exhaustive and further discussion is welcomed on the developer forums.
+
+
+## Principles
+
+### <a name="open_source"></a>Open Source
+> OpenLMIS is offered under an open source license, which means that everyone has the right to use and modify the 
+> software without paying a license fee. Changes and additions are made available to the community under the terms of 
+> the license via our code contribution process.
+
+OpenLMIS is built and licensed under an Open Source [license](LICENSE.txt).  In addition to the project being Open 
+Source, OpenLMIS strives to always be available to develop on, build, deploy, use and generally contribute to using 
+similarly licensed technologies.  In practice this means that strong preference is given to contributions and their 
+dependant technologies that are licensed similarly.  Contributions should aspire to contribute:
+
+* Code and other IP licensed in a compatible license as OpenLMIS.  Strong preference is given to the OpenLMIS license
+ for simplicity.
+* Dependencies on third-party libraries / tools should also be open source and freely distributable.
+
+### <a name="appropriate"></a>Appropriate
+> OpenLMIS is designed with a focus on users in low resource and capacity environments.  Representatives from these 
+> environments are welcomed and valued members of the community and their insights help shape the software.
+
+OpenLMIS is built and used by those in low-resource settings:
+
+* Internet is often slow and intermittent.  Features should be designed with these limitations in mind.  For example, most 
+work-flows should be optimized for slow internet and even work-flows with periods of non-connectivity.  
+Administrative screens however can often take shortcuts and assume that their users will have better internet 
+connectivity.
+* Processes not only vary and need to be configurable by program and implementation, they oftentimes are  
+used in parallel or supplement traditional paper processes.  Data collection and forms should strive to be able to 
+be configurable to match a paper form and be able to restore it historically.
+* Screens are often older and come with lower resolutions than the latest and greatest.  800x600 px screens are not uncommon.  Additionally, many work-flows that would be used by someone at the last mile will be used by someone with a
+smaller tablet or even a phone.
```

Comment:
My view is that OpenLMIS should favor and support work-flows with intermeitent connection/data as well as those that are completely off-line.  The latter is perhaps more debate-able though I see no reason not to hold it as a suggestion.  To me that's a core problem that OpenLMIS right now doesn't always support.  That's what I was trying to convey with that.  Weather it's a web UI or a mobile front-end I think comes after the question of weather or not the feature is designed to be used with questionable internet.  What can I do to clear that up?